src/string.cr

```diff
@@ -2027,11 +2027,11 @@ class String
 
   private def gsub_ascii_char(char, replacement)
     String.new(bytesize) do |buffer|
-      each_char_with_index do |my_char, i|
-        buffer[i] = if my_char == char
+      to_slice.each_with_index do |byte, i|
+        buffer[i] = if char === byte
```

Comment:
Please don't use  `===`, use `char.ord == byte`.

Reply:
Also i'd much prefer
```cr
if char.ord == byte
  buffer[i] = replacement.ord.to_u8
else
  buffer[i] = byte
end
```

Reply:
@RX14 I vote to change this formatter rule for this indent to uniformly used 2 spaces.

Reply:
That's another issue, and I disagree with you.

Reply:
Out of the curiosity, how's that you don't want to change the rule, while in the same time you suggest against using it? Is there sth I don't get?

Reply:
Because changing the rule would mean it looks like
```cr
buffer[i] = if char.ord == byte
  replacement.ord.to_u8
else
  byte
end
```

which i think is even more ugly than indenting it all.

My preference goes:
1. what I suggested (assign in the if)
2. what's implemented in this PR (indent the if)
3. what you suggested (don't indent the if)

Reply:
Got it! for my taste it'd be 3,1,2 - in that order.

Reply:
2. is what it currently looks like. Until now this PR doesn't change anything about that.

Reply:
well it won't get changed if it's not in this PR

Reply:
@RX14 now it is =)